packages/mason_cli/lib/src/commands/bundle.dart

```diff
@@ -46,15 +64,45 @@ class BundleCommand extends MasonCommand {
 
   @override
   Future<int> run() async {
-    if (results.rest.isEmpty) {
-      usageException('path to the brick template must be provided');
+    final source = results['source'] as String;
+
+    late final rest = results.rest.first;
+
+    final Brick brick;
+    if (source == 'git') {
+      if (results.rest.isEmpty) {
+        usageException('A repository url must be provided');
+      }
+      brick = Brick(
+        location: BrickLocation(
+          git: GitPath(
+            rest,
+            path: results['git-path'] as String?,
+            ref: results['git-ref'] as String?,
+          ),
+        ),
+      );
+    } else if (source == 'hosted') {
+      if (results.rest.isEmpty) {
+        usageException('A brick name must be provided');
+      }
+      brick = Brick(name: rest, location: const BrickLocation(version: 'any'));
+    } else {
+      if (results.rest.isEmpty) {
+        usageException('A path to the brick template must be provided');
+      }
+      brick = Brick(location: BrickLocation(path: rest));
     }
-    final brick = Directory(results.rest.first);
-    if (!brick.existsSync()) {
-      throw BrickNotFoundException(brick.path);
+
+    final String brickPath;
+    if (brick.location.isLocal) {
+      brickPath = brick.location.path!;
+    } else {
+      final cachedBrick = await globalBricksJson.add(brick);
```

Comment:
Save on global cache if it is a remote brick template.

Reply:
This will have the side-effect of making the bricks visible in `mason ls -g` not sure if that's desirable or not.

Reply:
Not desirable, changed to use systemp.